node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -37,20 +39,29 @@ object Main {
       LoggerFactory.getLogger(getClass).error("Unhandled exception in thread " + thread.getName, ex)
     })
 
-    val configuration = Configuration.build(args)
-    System.setProperty("rnode.data.dir", configuration.server.dataDir.toString) // NonUnitStatements
-
     implicit val scheduler: Scheduler = Scheduler.computation(
       Math.max(java.lang.Runtime.getRuntime.availableProcessors(), 2),
       "node-runner",
       reporter = UncaughtExceptionLogger
     )
 
+    val configuration = Configuration.build(args, decryptPrivateKey(console))
```

Comment:
This is main entry to RNode. Is this going to ask for password even on commands that don't need password.

Maybe it's better to push this check and console input inside Run command.

Reply:
I was trying to avoid putting `ConsoleIO`  into other packages which ends up with  not very nice organized code. I'll see if there are other ways to do that.